webconsole/src/main/java/org/apache/felix/webconsole/internal/core/BundlesServlet.java

```diff
@@ -1642,8 +1648,10 @@ private void installBundle( String location, File bundleFile, int startLevel, bo
                 Bundle[] bundles = BundleContextUtil.getWorkingBundleContext(this.getBundleContext()).getBundles();
                 for ( int i = 0; i < bundles.length; i++ )
                 {
+                    boolean isSameBSN = (bundles[i].getSymbolicName() != null && bundles[i].getSymbolicName().equals( symbolicName ));
```

Comment:
I think you can just do `symbolicName.equals(bundles[i].getSymbolicName())` as `symbolicName` can't be `null` here!

Reply:
BSN cannot be null according to Felix implementation but according to Javadoc, it can. I agree completely that the null check is superfluous here.

Reply:
It can't be null due to the if condition at https://github.com/apache/felix-dev/pull/1/files#diff-b66e715ba83980920efaa568f6819c45L1628!

Reply:
You could use `Objects.equals`, it handles `null` arguments.

Reply:
I appreciate the suggestions - but to be honest I just moved the check fro the if statement below out. I tend to change as little as possible to reduce the impact of anything going wrong. I suggest that this PR gets accepted and a subsequent cleanup commit would be made.

Reply:
I agree